widgetssdk/src/main/java/com/glia/widgets/config/deserializers/DpDeserializer.java

```diff
@@ -0,0 +1,24 @@
+package com.glia.widgets.config.deserializers;
+
+import com.glia.widgets.config.components.base.DpSize;
+import com.glia.widgets.config.components.base.DpSizeImpl;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+
+public class DpDeserializer implements JsonDeserializer<DpSize> {
+    @Override
+    public DpSize deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+        try {
+            float sizeDp = json.getAsFloat();
+            if (sizeDp <= 0.0f) return null;
```

Comment:
Not 100% sure but I think it should work even if it is `sizeDp <= 0` since left side is float so it should do the comparison in float as well.

Reply:
But you might want to test it before changing :)

Reply:
it will work, it will not work in kotlin, I just wrote automatically